rpc/server.go

```diff
@@ -60,6 +66,47 @@ func StartPluginAtPath(socketPath string, endpoint http.Handler) (chan<- struct{
 	return fromUser, stopped, nil
 }
 
+// Run the JSON RPC server listening at the given path
+func runJSONRPCServer(socketPath string, server *rpc.Server) (chan<- struct{}, <-chan error, error) {
+	listener, err := net.Listen("unix", socketPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	stop := make(chan struct{})
+	stopped := make(chan error)
+
+	userInitiated := new(bool)
+	go func() {
+		<-stop
+		*userInitiated = true
+		listener.Close()
+	}()
+
+	go func() {
+	serverLoop:
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				break serverLoop
```

Comment:
This may be personal taste, but i'd rather see an unqualified `break` here since the loop label isn't needed, or a loop control variable:

```go
listening := true
for listening {
  ...
  if err != nil {
    listening = false
    continue
  }
}
```

Though of these two, i lean towards the unqualified `break`.